src/__test__/AbstractAppliance.test.js

```diff
@@ -0,0 +1,96 @@
+import { AssertionError } from 'assert'
+import AbstractAppliance from '../AbstractAppliance'
+import {
+	AbstractInstantiationError,
+	NotImplementedError,
+} from '../errors'
+import Payload from '../Payload'
+import {
+	FullyImplementedAppliance,
+	PartiallyImplementedAppliance,
+} from './classes'
+
+describe('AbstractAppliance #unit', () => {
+	describe('construction', () => {
+		it('should throw an error when called directly', () => {
+			expect(() => {
+				new AbstractAppliance() // eslint-disable-line no-new
+			}).toThrow(AbstractInstantiationError)
+		})
+
+		it('should throw an error when getInputTypes() is called without implementation', () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
```

Comment:
If you want to save yourself some repetition, I believe this const declaration can be moved up into the `describe('construction')` scope.

Reply:
Will consider; that said, tests are a funny thing -- the best practice is to avoid shared setup / variables between tests (so there is no chance of contamination)

Reply:
Ah yes, this is a good point and I think a better practice. Disregard!